fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerEntityEvents.java

```diff
@@ -63,6 +65,28 @@ private ServerEntityEvents() {
 		}
 	});
 
+	/**
+	 * Called during {@link ItemEntity#tick()} on the logical server if an entity tries to pick up any {@link ItemEntity}.
+	 *
+	 * <p>Picking up of an item is determined by {@link ItemEntity#onPlayerCollision(PlayerEntity)}.
+	 *
+	 * <p>Returning {@code false} prevents vanilla from handling the pick-up.
+	 */
+	public static final Event<ItemPickup> ITEM_PICKUP = EventFactory.createArrayBacked(
```

Comment:
I have a strong belief that this should be in `ServerLivingEntityEvents` in the `fabric-entity-events-v1` module.